### PR TITLE
Bump Ormolu and Fourmolu to GHC-9.2-compatible versions

### PR DESCRIPTION
**Pull request by georgefst**
Related: #2503.

I haven't tested the Stack or Nix builds.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2579"></a>

### CODE REVIEW

**Review by jneira — COMMENTED** [review]
many thanks for updating both plugins
looks good to me, @pepeiborra what do you think about merge to master and then update the ghc-9.2.1 branch?

**Comment by michaelpj** [comment]
We could also merge the 9.2 branch (which it seems like we're about to do) and do this right afterwards.

**Comment by Ailrun** [comment]
Or merging this to ghc 9.2 branch is also an option

**Comment by jneira** [comment]
I would go for merging 9.2.1 first, there is lot of work already invested there and it almost has a working executable as is

**Comment by georgefst** [comment]
Yeah, I considered all of those options. I opened a separate PR because it looked like #2503 was going to be merged imminently.

**Comment by jneira** [comment]
tests are disabled for ghc-9.2.1 but should be ran, no?
removing merge me just in case

**Comment by jneira** [comment]
plugins were disabled in the cabal-ghc921.project so they were no being even built, it seems ormolu does not support ghc-lib-parser-9.2.1.*:

```
 [__4] rejecting: ghc-lib-parser-9.2.1.20220109 (conflict: ormolu =>
ghc-lib-parser>=9.0 && <9.1)
```

**Comment by jneira** [comment]
sorry i am starting to wonder if the goal of the pr was built both with ghc-9.2 or only bump versions to an intermmediate version :thinking:

**Comment by jneira** [comment]
After some cabal.project tweaks we reach a build error:

```
Building library for ormolu-0.3.1.0..
[ 1 of 47] Compiling GHC.DynFlags     ( src/GHC/DynFlags.hs, dist/build/GHC/DynFlags.o, dist/build/GHC/DynFlags.dyn_o )

src/GHC/DynFlags.hs:28:13: error:
Error:     Not in scope: ‘platformMini’
    Perhaps you meant one of these:
      ‘platformMisc’ (imported from GHC.Driver.Session),
      data constructor ‘PlatformMisc’ (imported from GHC.Settings),
      ‘platformMinInt’ (imported from GHC.Platform)
   |
28 |             platformMini =
   |             ^^^^^^^^^^^^

src/GHC/DynFlags.hs:29:15: error:
Error:     Not in scope: data constructor ‘PlatformMini’
    Perhaps you meant one of these:
      ‘PlatformMisc’ (imported from GHC.Settings),
      variable ‘sPlatformMisc’ (imported from GHC.Settings),
      variable ‘platformMisc’ (imported from GHC.Driver.Session)
   |
29 |               PlatformMini
   |               ^^^^^^^^^^^^

src/GHC/DynFlags.hs:30:19: error: Not in scope: ‘platformMini_arch’
   |
30 |                 { platformMini_arch = ArchUnknown,
   |                   ^^^^^^^^^^^^^^^^^

src/GHC/DynFlags.hs:31:19: error:
Error:     Not in scope: ‘platformMini_os’
    Perhaps you meant ‘platformMinInt’ (imported from GHC.Platform)
   |
31 |                   platformMini_os = OSUnknown
   |                   ^^^^^^^^^^^^^^^

src/GHC/DynFlags.hs:43:7: error:
Error:     Not in scope: ‘sPlatformConstants’
    Perhaps you meant one of these:
      data constructor ‘PlatformConstants’ (imported from GHC.Platform),
      ‘platformConstants’ (imported from GHC.Platform),
      ‘platform_constants’ (imported from GHC.Settings)
   |
43 |       sPlatformConstants =
   |       ^^^^^^^^^^^^^^^^^^

src/GHC/DynFlags.hs:44:28: error:
Error:     Not in scope: ‘pc_DYNAMIC_BY_DEFAULT’
   |
44 |         PlatformConstants {pc_DYNAMIC_BY_DEFAULT = False, pc_WORD_SIZE = 8},
   |                            ^^^^^^^^^^^^^^^^^^^^^
```

**Comment by georgefst** [comment]
There was just a bad merge from `master`, in a way that was very easy to miss. We didn't want to inherit the change in `cabal-ghc921.project` that ignored the local `hls-ormolu-plugin`. And we really don't want `allow-newer` - all the relevant bounds should already be correct.

**Review by michaelpj — APPROVED** [review]
Nice work!

**Comment by michaelpj** [comment]
Popping `merge-me` back on.

**Review by jneira — APPROVED** [review]
thanks!

File: haskell-language-server.cabal [hunk 1]
```diff
@@ -288,7 +288,7 @@ common floskell
     cpp-options: -Dfloskell
 
 common fourmolu
-  if flag(fourmolu) && (impl(ghc < 9.2.1) || flag(ignore-plugins-ghc-bounds))
+  if flag(fourmolu) && flag(ignore-plugins-ghc-bounds)
```
Comment (jneira) [inline haskell-language-server.cabal:291]: this should have been
```
if flag(fourmolu)
```
corrected in #2567 51db994

**Pull request merged**